torch/csrc/jit/codegen/cuda/lower_thread_predicate.h

```diff
@@ -2,41 +2,47 @@
 #include <torch/csrc/WindowsTorchApiMacro.h>
 
 #include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
 
 #include <bitset>
 
 namespace torch {
 namespace jit {
 namespace fuser {
 
-class TORCH_CUDA_API ThreadPredicates {
- private:
-  Fusion* fusion_;
+/*
+ * Map from tensorview to bit set represnting <BIDx, BIDy, BIDz, TIDx, TIDy,
+ * TIDz> If any dependency of TV had a parallelized reduction, we will track
+ * it here. This will be used for predicate generation to prevent
+ * parallelization on that axis. This is important if we have a reduction on
+ * for example TIDx, as the reduced value is only valid on threadIdx.x == 0
+ * therefore if we use that value later in the kernel we have that predicate.
+ * If we follow a reduction parallelized on TIDx with a broadcast on TIDx we
+ * no longer need the predicate and can reset the bit accordingly
+ */
+class TORCH_CUDA_API ThreadPredicateMap {
+ public:
+  using MapType =
+      std::unordered_map<const TensorView*, ir_utils::ParallelTypeBitmap>;
+  using const_iterator = MapType::const_iterator;
 
-  /*
-   * Map from tensorview to bit set represnting <BIDx, BIDy, BIDz, TIDx, TIDy,
-   * TIDz> If any dependency of TV had a parallelized reduction, we will track
-   * it here. This will be used for predicate generation to prevent
-   * parallelization on that axis. This is important if we have a reduction on
-   * for example TIDx, as the reduced value is only valid on threadIdx.x == 0
-   * therefore if we use that value later in the kernel we have that predicate.
-   * If we follow a reduction parallelized on TIDx with a broadcast on TIDx we
-   * no longer need the predicate and can reset the bit accordingly
-   */
-  std::unordered_map<const TensorView*, std::bitset<6>> thread_predicates;
+  ThreadPredicateMap(Fusion* _fusion);
```

Comment:
This constructor should be explicit.

Side note: I'm thinking that having the ThreadPredicateMap outside and independed of Fusion may not be the ideal interface: the lifetime of the predicates are closely tied to the Fusion (if the fusion changes the predicates would be invalid, so they are not really standalone). I think this is fine for this PR, but I think we should revisit it when we get a chance.

Reply:
explicit is added.

Reply:
Re the side note: Agreed.